integrations/clickup/main.py

```diff
@@ -0,0 +1,102 @@
+from enum import StrEnum
+from typing import Any, Dict
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from client import ClickUpClient
+
+
+class ObjectKind(StrEnum):
+    TEAM = "team"
+    PROJECT = "project"
+    ISSUE = "issue"
+
+
+def init_client() -> ClickUpClient:
+    return ClickUpClient(ocean.integration_config["clickup_token"])
+
+
+async def setup_application() -> None:
+    """Sets up the application, including creating webhooks."""
+    logic_settings = ocean.integration_config
+    logger.warning(logic_settings)
+    app_host = logic_settings.get("app_host")
+    if not app_host:
+        logger.warning(
+            "No app host provided, skipping webhook creation. "
+            "Without setting up the webhook, the integration will not export live changes from ClickUp"
+        )
+        return
+
+    clickup_client = init_client()
+    await clickup_client.create_events_webhook(
+        logic_settings["clickup_host"], app_host, logic_settings["team_id"]
+    )
+
+
+@ocean.on_resync(ObjectKind.TEAM)
+async def on_resync_teams(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    client = init_client()
+    async for teams in client.get_paginated_teams(
+        ocean.integration_config["clickup_host"]
+    ):
+        logger.info(f"Received team batch with {len(teams)} teams")
+        yield teams
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    client = init_client()
```

Comment:
Change the return type from ASYNC_GENERATOR_RESYNC_TYPE to RAW_RESULT. apply it to the project as well. this means you will not need `yield` instead of `return`